python/tests/test_highlevel.py

```diff
@@ -2145,10 +2146,38 @@ def test_complex_mutations(self):
         ts = tsutil.insert_branch_mutations(msprime.simulate(10, random_seed=1))
         self.verify_mutations(ts.first())
 
-    def test_str(self):
-        t = self.get_tree()
+    def test_str(self, ts_fixture):
+        t = ts_fixture.first()
         assert isinstance(str(t), str)
-        assert str(t) == str(t.get_parent_dict())
+        assert (
+            str(t)
+            == textwrap.dedent(
+                """
+            ╔═════════════════════════════╗
+            ║Tree                         ║
+            ╠═══════════════════╤═════════╣
+            ║Index              │        0║
+            ╟───────────────────┼─────────╢
+            ║Interval           │  0-1 (1)║
+            ╟───────────────────┼─────────╢
+            ║Roots              │        1║
+            ╟───────────────────┼─────────╢
+            ║Nodes              │       19║
+            ╟───────────────────┼─────────╢
+            ║Sites              │       11║
+            ╟───────────────────┼─────────╢
+            ║Mutations          │       11║
+            ╟───────────────────┼─────────╢
+            ║Total Branch Length│10.427971║
+            ╚═══════════════════╧═════════╝
+        """
+            )[1:]
+        )
+
+    def test_html_repr(self, ts_fixture):
+        html = ts_fixture.first()._repr_html_()
+        assert len(html) > 1900
+        assert "<tr><td>Total Branch Length</td><td>10.427971</td></tr>" in html
```

Comment:
Be nice to test if this is valid XML  as well, just pushing it through one of the parsers will be fine.

Reply:
Done